GVFS/GVFS.FunctionalTests/Tests/GitCommands/GitCommandsTests.cs

```diff
@@ -45,6 +46,75 @@ public void StatusTest()
+
+            string command = "status";
+            string controlRepoRoot = this.ControlGitRepo.RootPath;
+            string gvfsRepoRoot = this.Enlistment.RepoRoot;
+
+            environmentVariables["GIT_TRACE"] = expectedPath;
+            ProcessResult expectedResult = GitProcess.InvokeProcess(controlRepoRoot, command, environmentVariables);
+
+            environmentVariables["GIT_TRACE"] = actualPath;
+            ProcessResult actualResult = GitHelpers.InvokeGitAgainstGVFSRepo(gvfsRepoRoot, command, environmentVariables);
+
+            GitHelpers.ValidateGitOutput(command, expectedResult, actualResult);
+
+            this.FileSystem.FileExists(expectedPath).ShouldBeTrue();
+            this.FileSystem.FileExists(actualPath).ShouldBeTrue();
+
+            string lineToFind = "trace: built-in: git status";
+            this.FileSystem.ReadAllText(expectedResult.Output).IndexOf(lineToFind).ShouldBeAtLeast(0);
+            this.FileSystem.ReadAllText(actualResult.Output).IndexOf(lineToFind).ShouldBeAtLeast(0);
+        }
+
+        [TestCase]
+        public void StatusWithGitTraceToStderrTest()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+            environmentVariables["GIT_TRACE"] = "2";
+
+            string command = "status";
+            string controlRepoRoot = this.ControlGitRepo.RootPath;
+            string gvfsRepoRoot = this.Enlistment.RepoRoot;
+
+            ProcessResult expectedResult = GitProcess.InvokeProcess(controlRepoRoot, command, environmentVariables);
+            ProcessResult actualResult = GitHelpers.InvokeGitAgainstGVFSRepo(gvfsRepoRoot, command, environmentVariables);
+
+            GitHelpers.ValidateGitOutput(command, expectedResult, actualResult);
+
+            string.IsNullOrWhiteSpace(expectedResult.Errors).ShouldBeFalse();
+            string.IsNullOrWhiteSpace(actualResult.Errors).ShouldBeTrue();
+        }
+
+        [TestCase]
+        public void StatusWithGitTraceInvalidCharactersTest()
+        {
+            Dictionary<string, string> environmentVariables = new Dictionary<string, string>();
+
+            if (Path.GetInvalidPathChars().Length == 0)
+            {
+                return;
+            }
+
+            string tracePath = Path.Combine(this.Enlistment.EnlistmentRoot, $"trace-log{Path.GetInvalidPathChars()[0]}.txt");
+            environmentVariables["GIT_TRACE"] = tracePath;
+
+            string command = "status";
+            string controlRepoRoot = this.ControlGitRepo.RootPath;
+            string gvfsRepoRoot = this.Enlistment.RepoRoot;
+
+            ProcessResult expectedResult = GitProcess.InvokeProcess(controlRepoRoot, command, environmentVariables);
+            ProcessResult actualResult = GitHelpers.InvokeGitAgainstGVFSRepo(gvfsRepoRoot, command, environmentVariables);
+
+            GitHelpers.ValidateGitOutput(command, expectedResult, actualResult);

```

Comment:
Here you can just call `ValidateGitCommand` and remove 5 lines of code.  Makes me wonder if we can somehow get that method or add a method to handle `GIT_TRACE` environment variables and simplify these tests but that might be messier and if these are the only tests that will use it, doesn't make as much sense.

Reply:
In this one case, I can inject the environment into `ValidateGitCommand` and rely on its comparisons. That does save code overall, and makes it easier to add tests like it in the future.